### PR TITLE
[Merged by Bors] - Added NiFi 1.20.0 and 1.21.0 support

### PR DESCRIPTION
**Pull request by maltesander**
# Description

fixes https://github.com/stackabletech/nifi-operator/issues/403

- adapted examples
- adapted tests
- adapted docs
- adapted getting started guides

test: https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/nifi-operator-it-custom/32/

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,10 @@ All notable changes to this project will be documented in this file.
 
 ## [Unreleased]
 
+### Added
+
+- Added support for NiFi version 1.20.0 and 1.21.0 ([#464]).
```
Comment (sbernauer) [inline CHANGELOG.md:9]:
```suggestion
- Added support for NiFi versions 1.20.0 and 1.21.0 ([#464]).
```

File: docs/modules/nifi/examples/getting_started/getting_started.sh [hunk 2]
```diff
@@ -133,15 +133,15 @@ metadata:
...
+    productVersion: 1.21.0
     stackableVersion: "0.0.0-dev"
   clusterConfig:
-    listenerClass: external-unstable
     authentication:
       method:
         singleUser:
           adminCredentialsSecret: nifi-admin-credentials-simple
           autoGenerate: true
+    listenerClass: external-unstable
```
Comment (sbernauer) [inline docs/modules/nifi/examples/getting_started/getting_started.sh:144]: Whats the intention? Sort alphabetically? ;)

File: tests/test-definition.yaml [hunk 3]
```diff
@@ -18,24 +16,24 @@ dimensions:
...
-      - 1.18.0-stackable0.0.0-dev
+      - 1.21.0-stackable0.0.0-dev
   - name: ldap-use-tls
     values:
       - "false"
       - "true"
   - name: openshift
     values:
       - "false"
-  # Used for both, zookeeper and nifi
```
Comment (sbernauer) [inline tests/test-definition.yaml:38]: Why did you remove this comment? For me personally it helps

File: docs/modules/nifi/examples/getting_started/getting_started.sh (hunk 2 above)
Comment (maltesander) [reply docs/modules/nifi/examples/getting_started/getting_started.sh:144]: Si :)

File: tests/test-definition.yaml (hunk 3 above)
Comment (maltesander) [reply tests/test-definition.yaml:38]: ah i actually wanted to remove the number of tests and remove it from zookeeper but it would not have done that.... will put it back.

**Review by sbernauer — APPROVED** [review]

**Comment by maltesander** [comment]
bors merge
